### ISSUE TITLE
Why return `React.addons.cloneWithProps(container, ...)`

### ISSUE THREAD

**Issue by mikew** [voronianski/react-swipe#2]
When trying to use this component and passing a `ref` to props this causes a warning:

> Warning: You are calling cloneWithProps() on a child with a ref. This is dangerous because you're creating a new child which will not be added as a ref to its parent.

Changing the render function like so doesn't break anything and gets rid of the warning:

``` javascript
render: function() {
  this.props.style = styles.container

  var container = React.DOM.div(this.props,
    React.DOM.div({style: styles.wrapper},
      React.Children.map(this.props.children, function(child) {
        return React.addons.cloneWithProps(child, {style: styles.child})
      })
    )
  )

  return container
}
```

Unless there is a way to call `.next()` and `.prev()` without having to keep a `ref` around.

**Comment by jed** [comment]
i'd never seen this... perhaps this warning is new?

at any rate, i figured it was better not to mutate a props object. maybe a better way would be cloning the props object first?

**Comment by voronianski** [comment]
@mikew this issue is resolved in latest version - `2.0.3`.
